java/fury-core/src/main/java/org/apache/fury/config/FuryBuilder.java

```diff
@@ -303,7 +303,7 @@ private void finish() {
               + "`ClassResolver#setClassChecker`");
     }
     if (suppressClassRegistrationWarnings == null) {
-      suppressClassRegistrationWarnings = !requireClassRegistration;
+      suppressClassRegistrationWarnings = requireClassRegistration;
```

Comment:
This is just to set default value, we set it to true when ClassRegistration is disabled to reduce the verbose logs which may bother users.

Reply:
By default ClassRegistration is enabled, so the warning log is printed.

Should we only print the warning log when ClassRegistration is disabled?

So we should keep ClassRegistration and suppressClassRegistrationWarnings(suppressClassRegistrationWarnings is true, which means the warning log is not printed.) at the same value.

Reply:
How about set the `suppressClassRegistrationWarnings` to true by default. Users can set it to false if they want to do some security audit

Reply:
Looks good to me, updated.